plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtDeleteVMSnapshotCommandWrapper.java

```diff
@@ -58,6 +59,9 @@ public Answer execute(final DeleteVMSnapshotCommand cmd, final LibvirtComputingR
 
             snapshot = dm.snapshotLookupByName(cmd.getTarget().getSnapshotName());
 
+            s_logger.debug("Suspending domain " + vmName);
+            dm.suspend(); // suspend the vm to avoid image corruption
```

Comment:
should we do this conditionally and remember whether we did, to then only start it up again in the finally clause if it wasn't suspended in the first place?

Reply:
@weizhouapache can you wrap this in a state check?

Reply:
Will have to leave this for another time - otherwise, the "manually suspended" VM does indeed get's unsuspended - just managed to test it. WHY would someone in the first place even touch the VM manually, then expect ACS to handle it all?